exporter/dynatraceexporter/serialize_test.go

```diff
@@ -0,0 +1,105 @@
+)
+
+func Test_serializeIntGauge(t *testing.T) {
+	dp := pdata.NewIntDataPoint()
+	dp.SetValue(5)
+	dp.SetTimestamp(pdata.Timestamp(time.Date(2021, 07, 16, 12, 30, 0, 0, time.UTC).UnixNano()))
+
+	t.Run("with prefix and dimension", func(t *testing.T) {
+		got, err := serializeIntGauge("int_gauge", "prefix", dimensions.NewNormalizedDimensionList(dimensions.NewDimension("key", "value")), dp)
+		assert.NoError(t, err)
+		assert.Equal(t, "prefix.int_gauge,key=value gauge,5 1626438600000", got)
+	})
+}
+
+func Test_serializeGauge(t *testing.T) {
+	dp := pdata.NewNumberDataPoint()
+	dp.SetValue(5.5)
+	dp.SetTimestamp(pdata.Timestamp(time.Date(2021, 07, 16, 12, 30, 0, 0, time.UTC).UnixNano()))
+
+	t.Run("with prefix and dimension", func(t *testing.T) {
+		got, err := serializeGauge("dbl_gauge", "prefix", dimensions.NewNormalizedDimensionList(dimensions.NewDimension("key", "value")), dp)
+		assert.NoError(t, err)
+		assert.Equal(t, "prefix.dbl_gauge,key=value gauge,5.5 1626438600000", got)
+	})
+}
+
+func Test_serializeIntSum(t *testing.T) {
+	dp := pdata.NewIntDataPoint()
+	dp.SetValue(5)
+	dp.SetTimestamp(pdata.Timestamp(time.Date(2021, 07, 16, 12, 30, 0, 0, time.UTC).UnixNano()))
+
+	t.Run("delta with prefix and dimension", func(t *testing.T) {
+		got, err := serializeIntSum("int_sum", "prefix", dimensions.NewNormalizedDimensionList(dimensions.NewDimension("key", "value")), pdata.AggregationTemporalityDelta, dp)
+		assert.NoError(t, err)
+		assert.Equal(t, "prefix.int_sum,key=value count,delta=5 1626438600000", got)
+	})
+
+	t.Run("cumulative with prefix and dimension", func(t *testing.T) {
+		got, err := serializeIntSum("int_sum", "prefix", dimensions.NewNormalizedDimensionList(dimensions.NewDimension("key", "value")), pdata.AggregationTemporalityCumulative, dp)
+		assert.NoError(t, err)
+		assert.Equal(t, "prefix.int_sum,key=value count,5 1626438600000", got)
+	})
+}
+
+func Test_serializeSum(t *testing.T) {
+	dp := pdata.NewNumberDataPoint()
+	dp.SetValue(5.5)
+	dp.SetTimestamp(pdata.Timestamp(time.Date(2021, 07, 16, 12, 30, 0, 0, time.UTC).UnixNano()))
+
+	t.Run("delta with prefix and dimension", func(t *testing.T) {
+		got, err := serializeSum("double_sum", "prefix", dimensions.NewNormalizedDimensionList(dimensions.NewDimension("key", "value")), pdata.AggregationTemporalityDelta, dp)
+		assert.NoError(t, err)
+		assert.Equal(t, "prefix.double_sum,key=value count,delta=5.5 1626438600000", got)
+	})
+
+	t.Run("cumulative with prefix and dimension", func(t *testing.T) {
+		got, err := serializeSum("double_sum", "prefix", dimensions.NewNormalizedDimensionList(dimensions.NewDimension("key", "value")), pdata.AggregationTemporalityCumulative, dp)
+		assert.NoError(t, err)
+		assert.Equal(t, "prefix.double_sum,key=value count,5.5 1626438600000", got)
+	})
+}
+
+func Test_serializeHistogram(t *testing.T) {
+	dp := pdata.NewHistogramDataPoint()
+	dp.SetExplicitBounds([]float64{0, 2, 4, 8})
+	dp.SetBucketCounts([]uint64{0, 1, 0, 1, 0})
```

Comment:
Please also add a test where the first and last buckets are filled since that's a separate code path.